### DISCUSSION TITLE
Test mails are sent but then registration or password recovery emails are not sent

### DISCUSSION THREAD

**Discussion by robgitnyc** [mastodon/mastodon#21379] [Q&A] [answered]
Hi.

I just finished setting up my Mastodon instance. This is my smtp configuration:
```
SMTP_SERVER=mail.privateemail.com
SMTP_PORT=587
SMTP_LOGIN=info@mypersonaldomain.net
SMTP_PASSWORD=difficultpassword
SMTP_AUTH_METHOD=plain
SMTP_OPENSSL_VERIFY_MODE=none
SMTP_ENABLE_STARTTLS=auto
SMTP_FROM_ADDRESS='Admin <info@mypersonaldomain.net>'
```

 During the install process, when running ```RAILS_ENV=production bundle exec rake mastodon:setup```and when the app gave me the opportunity to test the smtp configuration (using namecheap's privateemail.com smtp servers), the test email was sent and received without problems. However, after the instance is up and running, it doesn't send registration or password recovery emails using the exact same, untouched configuration. 

This is what I can get from when running ```journalctl -u mastodon-*```
```
Nov 22 08:14:55 mypersonaldomain.net bundle[38734]: 2022-11-22T08:14:55.437Z pid=38734 tid=1mna WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"mailers","class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","wrapped":"ActionMailer::MailDeliveryJob","args":[{"job_class":"ActionMailer::MailDeliveryJob","job_id":"6ae95bc2-57f9-461f-b5f5-7e9564e54a30","provider_job_id":null,"queue_name":"mailers","priority":null,"arguments":["UserMailer","confirmation_instructions","deliver_now",{"args":[{"_aj_globalid":"gid://mastodon/User/4"},"33EqHsdG6i_paZcKxBas"],"_aj_ruby2_keywords":["args"]}],"executions":0,"exception_executions":{},"locale":"en","timezone":"UTC","enqueued_at":"2022-11-22T08:13:01Z"}],"jid":"c7ff1ad202f1b7b66e08df95","created_at":1669104781.517404,"enqueued_at":1669104891.6341236,"error_message":"535 5.7.8 Error: authentication failed: \n","error_class":"Net::SMTPAuthenticationError","failed_at":1669104785.5652897,"retry_count":2,"retried_at":1669104850.4733977}}
Nov 22 08:14:55 mypersonaldomain.net bundle[38734]: 2022-11-22T08:14:55.437Z pid=38734 tid=1mna WARN: Net::SMTPAuthenticationError: 535 5.7.8 Error: authentication failed:
Nov 22 08:14:55 mypersonaldomain.net bundle[38734]: 2022-11-22T08:14:55.437Z pid=38734 tid=1mna WARN: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:1015:in `check_auth_response'
Nov 22 08:14:55 mypersonaldomain.net bundle[38734]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:776:in `auth_plain'
Nov 22 08:14:55 mypersonaldomain.net bundle[38734]: /home/mastodon/.rbenv/versions/3.0.4/lib/ruby/3.0.0/net/smtp.rb:768:in `public_send'
```
Help will be much appreciated.

**Comment by robgitnyc** [comment, marked as answer]
Well solution was very tricky to find but very simple: just enclose login and password name with single quotes.

**Reply by nightpool** [reply]
Yes, special characters need to be quoted in .env files. this has nothing to do with SMTP

**Comment by NicolaeNMV** [comment]
Cool, yeah. "WARN: Net::SMTPAuthenticationError: 535 5.7.8 Error: authentication failed:" was quite self explanatory.
But there is a lack of documentation about smtp configuration, you're right.

**Comment by robgitnyc** [comment]
Hey guys @NicolaeNMV @nightpool, can you give me a hand here? Now the server is running and emails are sent, however when using the search function to find users from other instances i get a "500 internal server error" in the lower left corner. How can I fix it? Made a thread here https://github.com/mastodon/mastodon/discussions/21456
